api/api/handlers/product.py

```diff
@@ -76,20 +85,65 @@ def save_product(request):
     """ handles both update and create requests """
     data = request.json_body
     try:
-        product = request.context.get_product(request.subpath[0])
-    except InvalidId:
-        return {"message": "Invalid id given"}
-    except DoesNotExist:
+        validate(data, json.loads(request.context.product_schema))
+    except ValidationError:
+        return {"message": "invalid data"}
+    if request.subpath:
+        try:
+            product = request.context.get_product(request.subpath[0])
+        except DoesNotExist:
+            return {"message": "product not found"}
+    else:
         product = Hardware()
         product._id = ObjectId()
         product.category = request.context.name
 
     for field in data:
-        setattr(product, field, data[field])
+        new_field = field.replace(".", "")
+        setattr(product, new_field, data[field])
+
+    request.context.products.append(product)
     request.context.products.save()
     return {"message": "product saved"}
 
 
+@record_view(request_method="POST")
+def save_records(request):
+    """ handles the records requests """
+    data = request.json_body
+    for item in data['items']:
+        obj = json.loads(item)
```

Comment:
Since we were disregarding variable name comments you can ignore;

obj is far from descriptive.